core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
```

Comment:
@vanzin 
We can do the following: 
* Get ```SparkContext``` rid of the ```ExecutorAllocationClient```
* ```SparkContext``` still exposes the killExecutor(s) 
* Change ```ExecutorAllocationClient``` to return a list of filtered executors which were requested to be killed.

The current dev api for ```killExecutors``` returns a boolean. How do we want to interpret this. Because the ```CoarseGrainedSchedulerBackend``` can decide to kill a subset of the requested executors, so how do we interpret this? Killing any one of them should return true or killing all of them? The semantics are kind of hazy here. 

Can we change the api to return the list of killed executors making it consistent throughout? Or keep the api the same but have not very clear understanding.